### PR TITLE
[DNM] jewel: build/ops: fix some cmake problems

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
http://tracker.ceph.com/issues/18878

### CODE REVIEW

**Comment by Vicente-Cheng** [comment]
@tchaikov 

hi kefu: 
would you like to take some time to check this : )

thanks!!

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,15 +1,11 @@
 include(GetGitRevisionDescription)
 
 enable_language(C ASM)
-set(bindir ${CMAKE_INSTALL_PREFIX}/bin)
-set(sbindir ${CMAKE_INSTALL_PREFIX}/sbin)
-set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
-set(sysconfdir ${CMAKE_INSTALL_PREFIX}/etc)
-set(pkgdatadir ${CMAKE_INSTALL_PREFIX}/share)
```
Comment (tchaikov) [inline src/CMakeLists.txt]: please do not remove them at this moment, we use these variables for substitution in our .in files, for example, `src/ceph-crush-location.in` and `src/ceph-debugpack.in`. maybe we will clean them up in master by using the cmake variables. but not in jewel, where we just do the bug fixes.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -1,15 +1,11 @@
...
+include(GNUInstallDirs)
```
Comment (tchaikov) [inline src/CMakeLists.txt]: so the minimal change for the cmake issue this line, i think.

**Comment by tchaikov** [comment]
@Vicente-Cheng could you also fill the commit message with details explaining why these change are not cherry-picked from master?

**Comment by Vicente-Cheng** [comment]
@tchaikov 

hi kefu, I already done your reminder.
does this commit message with this cherry-pick looks ok?

**Comment by tchaikov** [comment]
@Vicente-Cheng maybe you can backport following commits instead?

- 7a602ec292b7e92c994d1718402e35a6c6692a02
- ce942bd619e542103d116deb694cc4b05f78f42f
- 11e426bf8b2df10fc7a9f34c8653bf4e666ec076

**Comment by Vicente-Cheng** [comment]
@dachary @tchaikov 
hi loicd, kefu:

sorry for updating late.

I already cherry-pick three commit that kefu suggests.
It looks good for cmake, but still have some problem for make (maybe mean make check)

something like rbd-mirror and other check maybe error, but it's enough for some components.
Although cmake is not ready on jewel, I would like to take some time for cmake on jewel.
how did you think? or just keep it (I mean don't bother with cmake in jewel)

thanks!!!

**Comment by tchaikov** [comment]
@Vicente-Cheng i'd suggest we can just live with it. it's nice to have cmake in jewel but it's not tested or supported in our CI infrastructure, so it's fragile. i'd recommend to focus on master.

**Comment by dachary** [comment]
I agree with @tchaikov, fixing cmake in jewel is going to be a very large effort if we want to do it right. I'm inclined to just forget about it.

**Comment by Vicente-Cheng** [comment]
@dachary @tchaikov thanks for quickly reply, so maybe this pr and #13355 do not be merged back to jewel. Should I close it?

**Comment by tchaikov** [comment]
regarding to this PR, i am good either way. but i don't want to waste your effort and this PR is not a risky change. so i am inclined to include it in jewel. but it's release team's call. @dachary what do you think?

**Comment by Vicente-Cheng** [comment]
@tchaikov @dachary 
hi kefu, loicd,

I just close #13355 because I am sure that pr should not be involve into jewel.
About this pr, I think I already got something good information about cmake.
If you think it doesn't need now,  just close it : )

thanks for your information and help!!
vicente

**Comment by smithfarm** [comment]
Prefer to leave jewel cmake as it is, since it is not used.

**Comment by Vicente-Cheng** [comment]
@smithfarm 

OK,  thank you Nathan!! : )

**Pull request closed without merging**